test/metabase/api/table_test.clj

```diff
@@ -124,6 +123,38 @@
                   (map #(select-keys % [:name :display_name :id :entity_type]))
                   set))))))
 
+(defmacro with-tables-as-uploads
```

Comment:
This felt like the lesser of evils versus coupling to the internals of the upload namespace to make actual uploads.

Once we expose a DELETE endpoint, the case could be made for creating and deleting them at an API level in this test, but I'm not convinced that added complexity and runtime is worth it.

Reply:
this also seems like a standard thing to do. tbh my biggest hesitation was "what if we start using tables that already have `is_upload` set to true and this macro sets it to `false` as pasrt of the cleanup?", but that's very much not the case now and if the future changes we should get obvious test failures from it.